core/src/main/scala/akka/persistence/dynamodb/internal/JournalDao.scala

```diff
@@ -124,8 +124,12 @@ import software.amazon.awssdk.services.dynamodb.model.Update
             .build()
         }.asJava
 
+      val firstEvent = events.head;
+      val token = s"${firstEvent.writerUuid}-${firstEvent.seqNr.toString}"
+
       val req = TransactWriteItemsRequest
         .builder()
+        .clientRequestToken(token)
```

Comment:
Looks like client request token has a max length of 36. So only enough for the UUID in default format.

Reply:
Hmmm... then the idempotence would be a problem... so I guess taking the last 36 characters of `${writerUuid}-${seqNr}` would work.

Reply:
Or we encode the UUID more efficiently. Say parse it from string and rewrite as base64. Then it would only take up 22 characters.